File: tests/fixtures/code-map/langs-ruby/defs.rb
require 'set'

MAX_SIZE = 100

module Widget
  VERSION = "1.0"

  class Config
    def initialize(name)
      @name = name
    end

    def describe
      @name
    end

    def self.build(name)
      new(name)
    end
  end

  def helper
    VERSION
  end
end

def top_level_fn(a)
  a + 1
end

File: tests/fixtures/code-map/langs-csharp/defs.cs
using System;

namespace Widget.Core;

public delegate int Transformer(int value);

public interface IStore
{
    void Save(string key);
}

public enum Color
{
    Red,
    Green
}

public struct Point
{
    public int X;
}

public record Money(decimal Amount);

public class Config : IStore
{
    public const int MaxSize = 100;
    private int _counter;
    public string Name { get; set; }

    public Config(string name)
    {
        Name = name;
    }

    public void Save(string key)
    {
    }

    public int Describe() => MaxSize;
}
